File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/library/items/media/files/playback/file/preparation/queues/QueueProviders.java
package com.lasthopesoftware.bluewater.client.library.items.media.files.playback.file.preparation.queues;


import com.vedsoft.lazyj.AbstractSynchronousLazy;
import com.vedsoft.lazyj.ILazy;

import java.util.Arrays;

public class QueueProviders {
	private static final ILazy<Iterable<IPositionedFileQueueProvider>> lazyProviders = new AbstractSynchronousLazy<Iterable<IPositionedFileQueueProvider>>() {
		@Override
		protected Iterable<IPositionedFileQueueProvider> initialize() throws Exception {
			return Arrays.asList(new CompletingFileQueueProvider(), new CyclicalFileQueueProvider());
		}
	};

	public static Iterable<IPositionedFileQueueProvider> providers() {
		return lazyProviders.getObject();
	}
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/playback/file/preparation/queues/QueueProviders.kt
package com.lasthopesoftware.bluewater.client.playback.file.preparation.queues

object QueueProviders {
    private val lazyProviders by lazy { listOf(CompletingFileQueueProvider(), CyclicalFileQueueProvider()) }

    fun providers(): Iterable<ProvidePositionedFileQueue> = lazyProviders
}
